search/house_numbers_matcher.cpp

```diff
@@ -1,249 +1,485 @@
+    "lsb",    "lt",      "m",       "mac",  "mad",  "mah",     "mak",     "mat",      "mb",
+    "mbb",    "mbn",     "mch",     "mei",  "mks",  "mm",      "mny",     "mo",       "mok",
+    "monica", "mor",     "morocco", "msb",  "mtj",  "mtk",     "mvd",     "n",        "na",
+    "ncc",    "ne",      "nij",     "nn",   "no",   "nr",      "nst",     "nu",       "nut",
+    "o",      "of",      "ofof",    "old",  "one",  "oo",      "opl",     "p",        "pa",
+    "pap",    "par",     "park",    "pav",  "pb",   "pch",     "pg",      "ph",       "phd",
+    "pkf",    "plaza",   "plot",    "po",   "pos",  "pp",      "pr",      "pra",      "pya",
+    "q",      "qq",      "quater",  "r",    "ra",   "rbo",     "rd",      "rear",     "reisach",
+    "rk",     "rm",      "ro",      "road", "rood", "rosso",   "rs",      "rw",       "s",
+    "sab",    "sal",     "sav",     "sb",   "sba",  "sbb",     "sbl",     "sbn",      "sbx",
+    "sc",     "sch",     "sco",     "seb",  "sep",  "sf",      "sgr",     "shop",     "sir",
+    "sj",     "sl",      "sm",      "sn",   "snc",  "so",      "som",     "south",    "sp",
+    "spi",    "spn",     "ss",      "st",   "sta",  "stc",     "std",     "stiege",   "street",
+    "suite",  "sur",     "t",       "tam",  "ter",  "terrace", "tf",      "th",       "the",
+    "tl",     "to",      "torre",   "tr",   "traf", "trd",     "ts",      "tt",       "tu",
+    "u",      "uhm",     "unit",    "utc",  "v",    "vi",      "vii",     "w",        "wa",
+    "way",    "we",      "west",    "wf",   "wink", "wrh",     "ws",      "wsb",      "x",
+    "xx",     "y",       "z",       "za",   "zh",   "zona",    "zu",      "zw",       "א",
+    "ב",      "ג",       "α",       "а",    "б",    "бб",      "бл",      "в",        "вл",
+    "вх",     "г",       "д",       "е",    "ж",    "з",       "и",       "к",        "л",
+    "лит",    "м",       "магазин", "н",    "о",    "п",       "р",       "разр",     "с",
+    "стр",    "т",       "тп",      "у",    "уч",   "участок", "ф",       "ц",        "ა",
+    "丁目",   "之",      "号",      "號",
+
+    // List of exceptions
+    "владение"
+};
+
+// Common strings in house numbers.
+// To get this list, just run:
+//
+// ./clusterize-tag-values.lisp house-number path-to-taginfo-db.db > numbers.txt
+// tail -n +2 numbers.txt  | head -78 | sed 's/^.*) \(.*\) \[.*$/"\1"/g;s/[ -/]//g;s/$/,/' |
+// sort | uniq
+const char * const g_patterns[] = {
+    "BL",  "BLN",  "BLNSL", "BN",   "BNL",    "BNSL", "L",   "LL",   "LN",  "LNL",  "LNLN", "LNN",
+    "N",   "NBL",  "NBLN",  "NBN",  "NBNBN",  "NBNL", "NL",  "NLBN", "NLL", "NLLN", "NLN",  "NLNL",
+    "NLS", "NLSN", "NN",    "NNBN", "NNL",    "NNLN", "NNN", "NNS",  "NS",  "NSN",  "NSS",  "S",
+    "SL",  "SLL",  "SLN",   "SN",   "SNBNSS", "SNL",  "SNN", "SS",   "SSN", "SSS",  "SSSS",
+};
+
+const char * const g_buildingSynonyms[] = {"building", "bldg",   "bld",  "bl",  "unit",  "block",
```

Comment:
Take a look at the ceiling - a lot of such synonyms here :) No methodology, a close inspection of taginfo.